transport/http2_client.go

```diff
@@ -542,6 +534,11 @@ func (t *http2Client) handleData(f *http2.DataFrame) {
 	data := make([]byte, size)
 	copy(data, f.Data())
 	s.write(recvMsg{data: data})
+
+	if f.FrameHeader.Flags.Has(http2.FlagDataEndStream) {
+		s.readClosed()
```

Comment:
The spec isn't clear on what the result status should be when trailers are omitted.  My interpretation was that the RPC should still be treated as successful.  Perhaps it would make sense to record the situation in the log either way.

In my case the server wasn't so much "broken" as just "partially implemented".